### PR TITLE
Develop Merge - Updates to TOT

### PR DESCRIPTION
**Pull request by kiritigowda**
* Zen DNN Samples
* MIGraphX Updates
* rocAL Bug Fixes
* Updates

### CODE REVIEW

**Review by asalmanp — APPROVED** [review]

**Comment by kiritigowda** [comment]
@LakshmiKumar23 some of the codacy issues are from rocAL, can you take a look at them.

File: rocAL/docs/examples/pytorch/test_training.py [hunk 1]
```diff
@@ -91,7 +91,8 @@ def prefetched_loader(loader, rocal_cpu):
 
         stream = torch.cuda.Stream()
         first = True
-
+        input = None
+        target = None
```
Comment (LakshmiKumar23) [inline rocAL/docs/examples/pytorch/test_training.py:95]: MLperf code. Ignore this warning

File: rocAL/rocAL_pybind/example/new_api/tf_petsTrainingExample/README.md [hunk 2]
```diff
@@ -1,8 +1,15 @@
 ## Running tf_petsTrainingExample
 
-For first run, to setup dataset, edit "train_withROCAL_withTFRecordReader.py" and set "DATASET_DOWNLOAD_AND_PREPROCESS = True"
-For subsequent runs, after the dataset has already been downloaded and preprocessed, set "DATASET_DOWNLOAD_AND_PREPROCESS = False"
+### Building the required TF Rocm docker
+Use the instructions in the [docker section](../../../../../docker) to build the required [Tensorflow docker](../../../../../docker/tensorflow)
+
+
+### Running the training
+
```
Comment (LakshmiKumar23) [inline rocAL/rocAL_pybind/example/new_api/tf_petsTrainingExample/README.md:8]: done

File: docker/tensorflow/mivisionx-with-tensorflow.dockerfile [hunk 3]
```diff
@@ -0,0 +1,57 @@
+FROM rocm/tensorflow:latest
+
```
Comment (LakshmiKumar23) [inline docker/tensorflow/mivisionx-with-tensorflow.dockerfile:2]: done

File: rocAL/rocAL_pybind/run.sh [hunk 4]
```diff
@@ -27,8 +23,14 @@ if [[ $# -eq 1 ]]; then
   echo "${blue}Running setup.py with $1 ${reset}"
 
   if [[ "$1" == "--backend_ocl" ]] || [[ "$1" == "--backend_hip" ]]; then
-    sudo "$DEFAULT_PYTHON" setup.py build $1
-    sudo "$DEFAULT_PYTHON" setup.py install $1
+    sudo "$DEFAULT_PYTHON" setup.py bdist_wheel $1
+    WHEEL_DIR="./dist"
```
Comment (LakshmiKumar23) [inline rocAL/rocAL_pybind/run.sh:27]: already double quotes. not sure what this is

File: rocAL/rocAL_pybind/run.sh [hunk 5]
```diff
@@ -27,8 +23,14 @@ if [[ $# -eq 1 ]]; then
...
+    WHEEL_NAME=""
+    for WHEEL_NAME in "$WHEEL_DIR"/*
+    do
+        echo "Going to install $WHEEL_NAME"
+    done
+    pip$PYTHON_VERSION install $WHEEL_NAME
   else
```
Comment (LakshmiKumar23) [inline rocAL/rocAL_pybind/run.sh:34]: same as above

**Pull request merged**